astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -151,51 +131,29 @@ class NDArithmeticMixin(object):
 
     Examples
     --------
+    Using this Mixin with `~astropy.nddata.NDData`:
+
+        >>> from astropy.nddata import NDData, NDArithmeticMixin
+        >>> class NDDataWithMath(NDArithmeticMixin, NDData):
+        ...     pass
+
+    Using it with one operand on an instance::
+
+        >>> ndd = NDDataWithMath(100)
+        >>> ndd.add(20)
+        NDDataWithMath(120)
+
+    Using it with two operand on an instance::
+
+        >>> ndd = NDDataWithMath(100)
+        >>> ndd.divide(1, 5)
```

Comment:
@mwcraig The example is probably not optimal, but one could do `ndd.divide(1, ndd)` in that case. Ok granted one can also do `NDDataWithMath(1, ndd)` to accomplish the reverse operation for the object.

It was really hard finding an implementation that worked and was easy to explain in the documentation (at least what arguments `operand` and `operand2` mean). It was quite easy to document the current behaviour that if one is given `self {op} operand` and if both are given `operand {op} operand2` is performed.